### ISSUE TITLE
use test.mat to predict the scores,but some scores are nan?

### ISSUE THREAD

**Issue by PuYuQian** [uci-cbcl/DanQ#13]
Hi,
recently I try to run the code and use the 'roc_curve' method of sklearn lib to complete the ROC  curve.But  when I use test.mat to predict the scores,I find some scores are nan as below.
[[  3.01186283e-05   1.29807665e-06   8.04380979e-05 ...,   1.28236379e-05
    1.28918364e-05   5.43438178e-03]
 [  3.00640259e-05   1.30328704e-06   8.17851032e-05 ...,   1.26049072e-05
    1.26921959e-05   5.40011935e-03]
 [             nan              nan              nan ...,           nan             nan              nan] ..., 
 [  3.92721742e-02   4.11137007e-03   3.05633545e-02 ...,   8.36040080e-02
    7.93742165e-02   7.82782771e-03]
 [  3.92719842e-02   4.11114981e-03   3.05634644e-02 ...,   8.36120248e-02
    7.93785900e-02   7.82747846e-03]
 [  3.92720513e-02   4.11104271e-03   3.05638630e-02 ...,   8.36143717e-02
    7.93796182e-02   7.82728661e-03]]
I don't know why this issue occurs,I load your provided bestmodel to be the weights.This issue caused the ROC-AUC to remain 0.5,it's too bad.So I want to know where my work is wrong.My code is as follows.
Thanks,
pu

`forward_lstm = LSTM(input_dim=320, output_dim=320, return_sequences=True)
backward_lstm = LSTM(input_dim=320, output_dim=320, return_sequences=True)
brnn = Bidirectional(forward=forward_lstm, backward=backward_lstm, return_sequences=True)

print 'building model'

model = Sequential()
model.add(Convolution1D(input_dim=4,
                        input_length=1000,
                        nb_filter=320,
                        filter_length=26,
                        border_mode="valid",
                        activation="relu",
                        subsample_length=1))

model.add(MaxPooling1D(pool_length=13, stride=13))

model.add(Dropout(0.2))

model.add(brnn)

model.add(Dropout(0.5))

model.add(Flatten())

model.add(Dense(input_dim=75*640, output_dim=925))
model.add(Activation('relu'))

model.add(Dense(input_dim=925, output_dim=919))
model.add(Activation('sigmoid'))

print 'compiling model'
model.compile(loss='binary_crossentropy', optimizer='rmsprop', class_mode="binary")

model.load_weights('data/DanQ_bestmodel.hdf5')

print 'loading test data'
testmat = scipy.io.loadmat('data/test.mat')
X, y = np.transpose(testmat['testxdata'],axes=(0,2,1)), testmat['testdata']

print 'predicting on test sequences'
test_predicts = model.predict(X)

print 'printing the ROC AUC'
fpr, tpr, thresholds = roc_curve(y[:,13], test_predicts[:,13]) 
roc_auc = auc(fpr, tpr)
plt.plot(fpr, tpr, lw=1, label='ROC(area = %0.2f)'% roc_auc )
plt.xlabel('False Positive Rate')
plt.ylabel('True Positive Rate')
plt.legend()
plt.show()`

**Comment by daquang** [comment]
Did you ever figure out what went wrong? My repository is pretty old. Still uses theano, and it was made before Bidirectional was natively available in keras. That may be part of the problem

**Comment by PuYuQian** [comment]
Hi, I am very happy to receive your reply. I don't solve it well,the loss is still nan, I think it is because  the version of theano and keras I use,including anaconda  is old and not proper,so I try to use PyTorch to complete DanQ and the result is optimizable. Do you think it's the version problem?

**Comment by daquang** [comment]
It is likely a version problem. Reproducing the results on PyTorch should be fairly trivial.
